include/simulation_lifecycle/utils/feature.h

```diff
@@ -0,0 +1,109 @@
+#ifndef __SIMULATION_LIFECYCLE_FEATURE_H__
+#define __SIMULATION_LIFECYCLE_FEATURE_H__
+
+#include "cJSON.h"
+#include "stdlib.h"
+#include "string.h"
+#include "simulation_lifecycle/structures.h"
+
+/**
+ * @brief Extracts a JSON representation of the features from a JSON representation of a geojson file.
+ * @param geo_json, a cJSON containing the JSON representation of a geojson file.
+ * @return a cJSON containing features. This value can be NULL.
+ */
+cJSON * geojson_get_features(cJSON * geo_json);
+
+/**
+ * @brief Extracts a JSON representation of the properties from a JSON representation of a feature.
+ * @param feature_json, a cJSON containing the JSON representation of a feature.
+ * @return a cJSON containing the properties of a feature. This value can be NULL.
+ */
+cJSON * feature_get_properties(cJSON * feature_json);
+
+/**
+ * @brief Extracts a JSON representation of an attribute property from a JSON representation of a feature.
+ * @param feature_json, a cJSON containing the JSON representation of a feature.
+ * @param property, a string containing the name of the attribute property to extract.
+ * @return a cJSON containing the value of the property. This value can be NULL.
+ */
+cJSON * feature_get_property(cJSON * feature_json, char * property);
+
+/**
+ * @brief Extracts an attribute property from a JSON representation of a feature as a string.
+ * @param feature_json, a cJSON containing the JSON representation of a feature.
+ * @param property, a string containing the name of the attribute property to extract.
+ * @return a string containing the value of the property. This value can be NULL.
+ */
+char * feature_get_string_property(cJSON * feature_json, char * property);
+
+/**
+ * @brief Extracts an attribute property from a JSON representation of a feature as an int.
+ * @param feature_json, a cJSON containing the JSON representation of a feature.
+ * @param property, a string containing the name of the attribute property to extract.
+ * @return a pointer to an int containing the value of the property. This value can be NULL.
+ */
+int * feature_get_int_property(cJSON * feature_json, char * property);
```

Comment:
It is weird to return a pointer to int. Why don't you just return the int?

Reply:
Reason is I wanted it to return a null to check if it was successful down the line. Returning a sentinel value, ie -1, is not good because that may ne the actual value of the field